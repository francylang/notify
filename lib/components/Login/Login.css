.login {
  text-align: center;
  color: #d35656; }

.login-form {
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  position: fixed;
  top: 50%;
  left: 50%;
  z-index: 100;
  transform: translate(-50%, -50%);
  background-color: #fff;
  height: 340px;
  width: 400px;
  border-radius: 5px;
  padding: 30px;
  margin: 40px 0 0;
  box-shadow: 5px 10px 25px #bdbdbd;
  overflow: hidden;
  animation: appear 0.2s linear; }

input {
  width: 100%;
  background-color: transparent;
  border: none;
  border-bottom: 1px solid #bdbdbd;
  font-size: 20px;
  font-weight: lighter;
  margin: 20px 0; }

input:focus {
  outline: none;
  border-bottom: 2px solid #d35656; }

.login-submit-button {
  background-color: #d35656;
  border: none;
  border-radius: 10px;
  color: #fff;
  padding: 8px 20px;
  font-size: 16px;
  margin: 20px 0;
  width: 100%;
  cursor: pointer; }

.login-submit-button:hover {
  background-color: #E65C4F; }

.login-submit-button:disabled {
  background-color: #bdbdbd; }
