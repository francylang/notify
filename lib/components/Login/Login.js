import React, { Component } from 'react';
import PropTypes from 'prop-types';
import userLoginData from '../../utils/userLoginData.js';


// import { Redirect } from 'react-router';


class Login extends Component {
  constructor() {
    super();
    this.state = {
      email: '',
      password: ''
    };
    this.handleSubmit = this.handleSubmit.bind(this);
  }

  handleChange(event, type) {
    let value = event.target.value;

    this.setState({
      [type]: value
    });
  }

  handleSubmit(event) {
    event.preventDefault();
    const user = this.state;
    this.props.isAuthenticated()
    const realUser = user === userLoginData ? true : false
    console.log(userLoginData);
    console.log(realUser);
    console.log(user);
    this.setState({
      email: '',
      password: ''
    });
    // this.props.history.push('/');
  }

  render() {
    const isDisabled =
    this.state.email.length &&
    this.state.password.length ? false : true;

    return (
      <div>
        <form className='login-form'>
          <h2 className='login'>Login</h2>
            <input
              type='text'
              placeholder='Email'
              value={this.state.email}
              onChange={ (event) => this.handleChange(event, 'email')}>
            </input>
            <input
              type='text'
              placeholder='Password'
              value={this.state.password}
              onChange={ (event) => this.handleChange(event, 'password')}>
            </input>
            <input
              className='login-submit-button'
              disabled={isDisabled}
              type='submit'
              onClick={this.handleSubmit}
              >
            </input>
        </form>
      </div>
    );
  }
}

Login.propTypes = {
  history: PropTypes.object
};


export default Login;
