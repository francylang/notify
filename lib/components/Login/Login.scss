$main-color: #1a90bf ;
$main-color-hover: #405780;

.login {
  text-align: center;
  color: $main-color;
  font-size: 35px;
}

.login-form {
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  position: fixed;
  top: 50%;
  left: 50%;
  z-index: 100;
  transform: translate(-50%, -50%);
  background-color: #fff;
  height: 400px;
  width: 500px;
  border-radius: 5px;
  padding: 30px;
  margin: 40px 0 0;
  box-shadow: 5px 10px 25px #bdbdbd;
  overflow: hidden;
  animation: appear 0.2s linear;
}

input {
  width: 100%;
  background-color: transparent;
  border: none;
  border-bottom: 1px solid #bdbdbd;
  font-size: 20px;
  font-weight: lighter;
  margin: 20px 0;
}
input:focus {
  outline: none;
  border-bottom: 2px solid $main-color;
}

.login-submit-button {
  background-color: $main-color;
  border: none;
  border-radius: 10px;
  color: #fff;
  padding: 10px 20px;
  font-size: 25px;
  margin: 20px 0;
  width: 100%;
  cursor: pointer;
}

.login-submit-button:hover {
    background-color: $main-color-hover;
}

.login-submit-button:disabled {
    background-color: #bdbdbd;
}

@media screen and (max-width: 500px) {
  .login {
    font-size: 23px;
  }

  .login-form {
    height: 348px;
    width: 260px;
    margin-top: 100px;
  }

  .login-submit-button,
  .login-submit-button:hover,
  .login-submit-button:disabled {
    margin: 0 0 42px 0;
    padding: 8px 20px;
    font-size: 23px;
  }

}
