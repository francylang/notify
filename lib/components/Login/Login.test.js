import React from 'react';
import Login from './Login.js';
import { shallow } from 'enzyme';

describe('<Login />', () => {

  it('should exist', () => {
    const wrapper = shallow(<Login />);

    expect(wrapper).toBeDefined();
  });

  it('should match the Login snapshot', () => {
    let wrapper = shallow(<Login />);

    expect(wrapper).toMatchSnapshot();
  });
});
