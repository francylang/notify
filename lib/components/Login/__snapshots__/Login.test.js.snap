// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Login /> should match the Login snapshot 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div>
    <form
        className="login-form"
    >
        <h2
            className="login"
        >
            Login
        </h2>
        <input
            onChange={[Function]}
            placeholder="Email"
            type="text"
            value=""
        />
        <input
            onChange={[Function]}
            placeholder="Password"
            type="text"
            value=""
        />
        <input
            className="login-submit-button"
            disabled={true}
            onClick={[Function]}
            type="submit"
        />
    </form>
</div>,
  "nodes": Array [
    <div>
      <form
            className="login-form"
      >
            <h2
                  className="login"
            >
                  Login
            </h2>
            <input
                  onChange={[Function]}
                  placeholder="Email"
                  type="text"
                  value=""
            />
            <input
                  onChange={[Function]}
                  placeholder="Password"
                  type="text"
                  value=""
            />
            <input
                  className="login-submit-button"
                  disabled={true}
                  onClick={[Function]}
                  type="submit"
            />
      </form>
</div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <Login />,
      "_debugID": 3,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": Login {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "handleSubmit": [Function],
        "props": Object {},
        "refs": Object {},
        "state": Object {
          "email": "",
          "password": "",
        },
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 2,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div>
          <form
                    className="login-form"
          >
                    <h2
                              className="login"
                    >
                              Login
                    </h2>
                    <input
                              onChange={[Function]}
                              placeholder="Email"
                              type="text"
                              value=""
                    />
                    <input
                              onChange={[Function]}
                              placeholder="Password"
                              type="text"
                              value=""
                    />
                    <input
                              className="login-submit-button"
                              disabled={true}
                              onClick={[Function]}
                              type="submit"
                    />
          </form>
</div>,
        "_debugID": 4,
        "_renderedOutput": <div>
          <form
                    className="login-form"
          >
                    <h2
                              className="login"
                    >
                              Login
                    </h2>
                    <input
                              onChange={[Function]}
                              placeholder="Email"
                              type="text"
                              value=""
                    />
                    <input
                              onChange={[Function]}
                              placeholder="Password"
                              type="text"
                              value=""
                    />
                    <input
                              className="login-submit-button"
                              disabled={true}
                              onClick={[Function]}
                              type="submit"
                    />
          </form>
</div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Login />,
}
`;
